src/components/organisms/DataTable/DataTable.tsx

```diff
@@ -105,24 +108,14 @@ const DataTable = <TData,>(
     getRowStyles,
     columnOrder,
     subRowComponent,
+    pagination,
+    setPagination,
   }: DataTableProps<TData>,
   ref: Ref<HTMLDivElement>
 ) => {
   const [horizontalWrapperId] = useState(useId())
   const { per_page, last_page } = paginationData || {}
   const [expanded, setExpanded] = useState<ExpandedState>({})
-  const [pagination, setPagination] = useState<PaginationState>({
```

Comment:
**[blocker]** (Possibly)

I would prefer a less "intrusive" solution, where we change far less things. The general logic is that if filters change, then reset pagination. As the pagination data for every BE request comes from/uses our useFilters hook, it would be a very good place where to handle it. 

Basically inside the "handleFilterChange" and "handleSortingChange" functions we could just call handlePaginationChange({ page: 0 }) (since the per_page value shouldn't change). And we would get away with essentially 2 added lines in 1 file + Possibly handling this manually, if we have any exceptions somewhere, that don't use the useFilters helper hook

Reply:
Couldn't use handlePaginationChange there since the handle methods seem to overwrite each other, but I found a workaround and the result is definitely much less intrusive. Take look when you can and let me know what you think!